modules/apps/object/object-entry-folder-rest-impl/rest-openapi.yaml

```diff
@@ -0,0 +1,405 @@
+                200:
+                    content:
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/ObjectEntryFolder"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/ObjectEntryFolder"
+            tags: ["ObjectEntryFolder"]
+        patch:
+            description:
+                Updates only the fields received in the request body, leaving any other fields untouched.
+            operationId: patchObjectEntryFolder
+            parameters:
+                - in: path
+                  name: objectEntryFolderId
+                  required: true
+                  schema:
+                      format: int64
+                      type: integer
+            requestBody:
+                content:
+                    application/json:
+                        schema:
+                            $ref: "#/components/schemas/ObjectEntryFolder"
+                    application/xml:
+                        schema:
+                            $ref: "#/components/schemas/ObjectEntryFolder"
+            responses:
+                200:
+                    content:
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/ObjectEntryFolder"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/ObjectEntryFolder"
+            tags: ["ObjectEntryFolder"]
+        put:
+            description:
+                Replaces the object entry folder with the information sent in the request body. Any missing fields are
+                deleted, unless they are required.
+            operationId: putObjectEntryFolder
+            parameters:
+                - in: path
+                  name: objectEntryFolderId
+                  required: true
+                  schema:
+                      format: int64
+                      type: integer
+            requestBody:
+                content:
+                    application/json:
+                        schema:
+                            $ref: "#/components/schemas/ObjectEntryFolder"
+                    application/xml:
+                        schema:
+                            $ref: "#/components/schemas/ObjectEntryFolder"
+            responses:
+                200:
+                    content:
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/ObjectEntryFolder"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/ObjectEntryFolder"
+            tags: ["ObjectEntryFolder"]
+    "/scopes/{scopeKey}/object-entry-folder/by-external-reference-code/{externalReferenceCode}":
```

Comment:
Everywhere you have an `scopeKey` (ObjectDefinitionConstants.SCOPE_*) you'll need an `scopeId` (group or company ID). For object entries it is not necessary because its object definition is always known (and that provides the scope ID).

Reply:
ScopeKey : We took the naming of rest API to standardise.

<img src="https://media3.giphy.com/media/v1.Y2lkPWJkM2VhNTdldWd4aHpwdzRveXE1b2I4d2t5cTgwMGRuZGUxeTdrY2Y3YTNiM2ZzNSZlcD12MV9naWZzX3NlYXJjaCZjdD1n/Ld77zD3fF3Run8olIt/giphy.gif"/>


you want to change it?

Reply:
The naming is good; what needs to be done is to add the scope ID in addition to the key. Something like:
```
/scopes/{scopeKey}/{scopeId}/object-entry-folders/...
```
For object entry the scopeId is not necessary because the ObjectDefinition is injected in the resources (so the scope ID is always implicit), but for object entry folders we won't have that, so we need to pass it explicitly.

Reply:
we do not understand this. Can we talk?